test/index.spec.ts

```diff
@@ -137,452 +137,540 @@ test('encryptor:encrypt & decrypt with wrong password', async ({ page }) => {
+        JSON.stringify(data),
+      );
+    });
+
+    test('encryptor:decryptWithDetail works with password after encryption with key', async ({
+      page,
+    }) => {
+      const password = 'a sample passw0rd';
+      const startingData = { foo: 'data to encrypt' };
+
+      // Get an exported key to use
+      const { salt, exportedKeyString } = await page.evaluate(
+        async (args) => {
+          const usedSalt = window.encryptor.generateSalt();
+          const { exportedKeyString: newKeyString } =
+            await window.encryptor.encryptWithDetail(
+              args.password,
+              args.data,
+              usedSalt,
+            );
+
+          return {
+            salt: usedSalt,
+            exportedKeyString: newKeyString,
+          };
+        },
+        { data: startingData, password },
+      );
 
-  // Update the data, encrypt using key
-  const newData = { ...startingData, bar: 'more data' };
-  const encryptWithKeyResult = await page.evaluate(
-    async (args) => {
-      const key = await window.encryptor.importKey(args.keyString);
-      const result = await window.encryptor.encryptWithKey(key, args.data);
+      // Update the data, encrypt using key
+      const newData = { ...startingData, bar: 'more data' };
+      const encryptWithKeyResult = await page.evaluate(
+        async (args) => {
+          const key = await window.encryptor.importKey(args.keyString);
+          return await window.encryptor.encryptWithKey(key, args.data);
+        },
+        { data: newData, keyString: exportedKeyString },
+      );
 
-      return {
-        encryptWithKeyResult: result,
-        exportedKeyString: await window.encryptor.exportKey(key),
+      // Mock the encrypted object
+      const decryptable = {
+        ...encryptWithKeyResult,
+        salt,
       };
-    },
-    { data: newData, keyString: exportedKeyString },
-  );
 
-  // Prove that a vault created with key can be decrypted with password
-  const decryptedResult = await page.evaluate(
-    async (args) => {
-      const key = await window.encryptor.importKey(args.exportedKeyString);
-      return await window.encryptor.decryptWithKey(key, args.data);
-    },
-    {
-      exportedKeyString: encryptWithKeyResult.exportedKeyString,
-      data: encryptWithKeyResult.encryptWithKeyResult,
-    },
-  );
+      // Prove that a vault created with key can be decrypted with password
+      const decryptedResult = await page.evaluate(
+        async (args) =>
+          await window.encryptor.decryptWithDetail(args.password, args.data),
+        { password, data: JSON.stringify(decryptable) },
+      );
 
-  expect(JSON.stringify(decryptedResult)).toStrictEqual(
-    JSON.stringify(newData),
-  );
-});
+      expect(JSON.stringify(decryptedResult.vault)).toStrictEqual(
+        JSON.stringify(newData),
+      );
+    });
+
+    test('encryptor:encryptWithKey works with decryptWithKey', async ({
```

Comment:
This test doesn't rely on either `sampleEncryptedData` or `oldSampleEncryptedData`. Should it be inside the loop?